python/src/iceberg/expressions/base.py

```diff
@@ -122,3 +124,37 @@ def __le__(self, other):
 
     def __ge__(self, other):
         return self.value >= other.value
+
+
+class Accessor(Generic[T], ABC):
+    """An accessor for a specific position in a container of type T
```

Comment:
Do we need `T`? We parameterize that type in Java, but because types are more rigid. In Python, we can probably use the `StructProtocol` instead because a class can "implement" the protocol just by having those methods.

Reply:
You're right, I think I was under the impression that we'd have implementations of this class but this is a class that will be instantiated and used directly right? Something like:

```py
container = FooStruct()
container.set(0, "foo")

value = Accessor(position=0, iceberg_type=StringType).get(container)
print(value)  # "foo"
```
Does that look right?

Reply:
It's really close. In Java, we pass in the JVM class for type safety since this returns the class that you pass in. In Python, there's no need. So you don't need the `iceberg_type` parameter at all when creating it. Otherwise it looks good.

Reply:
Changed `Accessor` to a concrete class. 👍